test/plugin/base_test.rb

```diff
@@ -536,6 +536,57 @@ def test_structured_payload_json_log_detect_json_not_parsed_json
     end
   end
 
+  def test_structured_payload_json_log_detect_json_not_parsed_log_msg_hash
```

Comment:
Do we want to also test that this happens when `detect_json` is false (i.e., the field is left alone)?

Reply:
Added.